### PR TITLE
feat: use updated linter and strict typing

### PR DESCRIPTION
**Pull request by acezard**
This merge request allows us to use the latest eslint features, which is especially important for Typescript. 

In order to make it work, it was necessary to use a fork of cozy-scripts, but also to update the dev dependencies of cozy-notes itself. This is intended as a temporary but necessary change. Indeed, it is difficult to work without proper linting of Typescript files.

It should not in any way introduce regressions in the app itself, because it doesn't touch anything runtime-related. If anything, this configuration will allow us to spot issues before they are delivered to the master branch.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,7 +67,7 @@
     "cozy-editor-core": "134.0.3",
     "cozy-flags": "2.7.2",
     "cozy-realtime": "3.13.0",
-    "cozy-scripts": "5.10.0",
+    "cozy-scripts": "npm:anc-scripts@^6.0.0",
```
Comment (Crash--) [inline package.json:70]: Nein. We don't use personal fork. If you want to use a new version of cozy-scripts or else, you can publish a new tag for our cozy-scripts lib and use it here. You can create a `cozy-script@eslint7" or something

**Review by Crash-- — CHANGES_REQUESTED** [review]
see comment

File: package.json (hunk 1 above)
Comment (acezard) [reply package.json:70]: Yes I was expecting that, will do

Comment (acezard) [reply package.json]: updated

**Review by Crash-- — APPROVED** [review]

**Pull request merged**